### ISSUE TITLE
Running problem

### ISSUE THREAD

**Issue by JhinAir** [zhqingit/giremi#51]
Hi, I got such problem as follow:

*** Error in `/home/user/liu/Software/giremi-master/giremi': free(): invalid next size (normal): 0x00000000a1814770 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81499)[0x7fd20a236499]
/home/user/liu/Software/giremi-master/giremi[0x41f02d]
/home/user/liu/Software/giremi-master/giremi[0x41f52e]
/home/user/liu/Software/giremi-master/giremi[0x411703]
/home/user/liu/Software/giremi-master/giremi[0x402165]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7fd20a1d7445]
/home/user/liu/Software/giremi-master/giremi[0x401c79]
======= Memory map: ========
00400000-00425000 r-xp 00000000 00:29 76411379                           /home/user/liu/Software/giremi-master/giremi
00624000-00625000 r--p 00024000 00:29 76411379                           /home/user/liu/Software/giremi-master/giremi
00625000-00626000 rw-p 00025000 00:29 76411379                           /home/user/liu/Software/giremi-master/giremi
01791000-a181a000 rw-p 00000000 00:00 0                                  [heap]
7fd200000000-7fd200021000 rw-p 00000000 00:00 0
7fd200021000-7fd204000000 ---p 00000000 00:00 0
7fd20558c000-7fd20758d000 rw-p 00000000 00:00 0
7fd208b75000-7fd208b8a000 r-xp 00000000 08:03 951696                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7fd208b8a000-7fd208d89000 ---p 00015000 08:03 951696                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7fd208d89000-7fd208d8a000 r--p 00014000 08:03 951696                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7fd208d8a000-7fd208d8b000 rw-p 00015000 08:03 951696                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7fd208d8f000-7fd209d90000 rw-p 00000000 00:00 0
7fd209f95000-7fd209fac000 r-xp 00000000 08:03 918504                     /usr/lib64/libpthread-2.17.so
7fd209fac000-7fd20a1ab000 ---p 00017000 08:03 918504                     /usr/lib64/libpthread-2.17.so
7fd20a1ab000-7fd20a1ac000 r--p 00016000 08:03 918504                     /usr/lib64/libpthread-2.17.so
7fd20a1ac000-7fd20a1ad000 rw-p 00017000 08:03 918504                     /usr/lib64/libpthread-2.17.so
7fd20a1ad000-7fd20a1b1000 rw-p 00000000 00:00 0
7fd20a1b5000-7fd20a378000 r-xp 00000000 08:03 918478                     /usr/lib64/libc-2.17.so
7fd20a378000-7fd20a577000 ---p 001c3000 08:03 918478                     /usr/lib64/libc-2.17.so
7fd20a577000-7fd20a57b000 r--p 001c2000 08:03 918478                     /usr/lib64/libc-2.17.so
7fd20a57b000-7fd20a57d000 rw-p 001c6000 08:03 918478                     /usr/lib64/libc-2.17.so
7fd20a57d000-7fd20a582000 rw-p 00000000 00:00 0
7fd20a585000-7fd20a614000 r-xp 00000000 00:29 76622768                   /home/user/liu/Software/lib/lib/libhts.so.1.3
7fd20a614000-7fd20a813000 ---p 0008f000 00:29 76622768                   /home/user/liu/Software/lib/lib/libhts.so.1.3
7fd20a813000-7fd20a814000 r--p 0008e000 00:29 76622768                   /home/user/liu/Software/lib/lib/libhts.so.1.3
7fd20a814000-7fd20a816000 rw-p 0008f000 00:29 76622768                   /home/user/liu/Software/lib/lib/libhts.so.1.3
7fd20a81d000-7fd20a91e000 r-xp 00000000 08:03 954340                     /usr/lib64/libm-2.17.so
7fd20a91e000-7fd20ab1d000 ---p 00101000 08:03 954340                     /usr/lib64/libm-2.17.so
7fd20ab1d000-7fd20ab1e000 r--p 00100000 08:03 954340                     /usr/lib64/libm-2.17.so
7fd20ab1e000-7fd20ab1f000 rw-p 00101000 08:03 954340                     /usr/lib64/libm-2.17.so
7fd20ab25000-7fd20ab3a000 r-xp 00000000 08:03 918928                     /usr/lib64/libz.so.1.2.7
7fd20ab3a000-7fd20ad39000 ---p 00015000 08:03 918928                     /usr/lib64/libz.so.1.2.7
7fd20ad39000-7fd20ad3a000 r--p 00014000 08:03 918928                     /usr/lib64/libz.so.1.2.7
7fd20ad3a000-7fd20ad3b000 rw-p 00015000 08:03 918928                     /usr/lib64/libz.so.1.2.7
7fd20ad3d000-7fd20ad5f000 r-xp 00000000 08:03 918467                     /usr/lib64/ld-2.17.so
7fd20af3b000-7fd20af3d000 rw-p 00000000 00:00 0
7fd20af59000-7fd20af5e000 rw-p 00000000 00:00 0
7fd20af5e000-7fd20af5f000 r--p 00021000 08:03 918467                     /usr/lib64/ld-2.17.so
7fd20af5f000-7fd20af60000 rw-p 00022000 08:03 918467                     /usr/lib64/ld-2.17.so
7fd20af60000-7fd20af61000 rw-p 00000000 00:00 0
7fff07cc9000-7fff07cec000 rw-p 00000000 00:00 0                          [stack]
7fff07cfd000-7fff07cff000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
/var/spool/slurm/slurmd/job4587259/slurm_script: line 11: 26894 Aborted                 giremi -f dmel.fasta -l Dmel.DB.SNPlist -m 5 -p 1 -s 0 -o Dmel.RNAediting D.mel_carcass.bam 

I set a big memmory as 500Gb, but it still gets problem. Could you help solve this?
Best~
Jing

**Comment by JhinAir** [comment]
It's a bug, I think.

**Comment by JhinAir** [comment]
Yeah, the above issue has been resolved (actually not really) as the software seems be not able to handle too much SNPs. I removed homozygous SNP (otherwise you will get errors in the r), and of course came across the never resolved problem.

Error in data.frame(..., check.names = FALSE) :
  arguments imply differing number of rows: 0, 1
Calls: cbind -> cbind -> data.frame
Execution halted

I looked into the giremi.r and I found these defined values are all zero, causing this error.
#########
> dat[rownames(subset(dat,pvalue_MI>0 & pvalue_MI<=0.05 & ifSNP==0)),"ifRNAE"]= 1
> dat.pos <- subset(dat, pvalue_MI>0 & pvalue_MI <= 0.01 & ifSNP==0 & RNAE_t == "AG")
> if (nrow(dat.pos)==0){
+ dat.pos <- subset(dat, pvalue_MI>0 & pvalue_MI <= 0.05 & ifSNP==0 & RNAE_t == "AG")
+ }
> dat.neg <- subset(dat,ifNEG == 1)
#########

I really doubt about the usability of this software. I don't want to be a bad guy, but just annoyed of wasting much time. Maybe the software can work well with unexplored process, but the author would never write out in the readme.

Hope good luck with the later.

**Comment by xyxss** [comment]
Same problem. This step filtered all sites.

**Comment by jumphone** [comment]
**Advertisement**
If you want to detect RNA editing sites, please try **SPRINT**:
https://github.com/jumphone/SPRINT
